### PR TITLE
VP-6068: Complete cart authentication authorization scenarios

### PR DESCRIPTION
**Pull request by AliveMen**
### Related task:     
    VP-6068 Complete cart authentication/authorization scenarios
### Problem:
   Used authorization for single cart query. Insecurity for search cart query 
### Solution:
   Rework authorization for cart queries 
### Changes:
   Remove authorization from query for single cart 
   Added authorization for search cart query

### CODE REVIEW

File: src/XPurchase/VirtoCommerce.XPurchase/Schemas/PurchaseSchema.cs [hunk 1]
```diff
@@ -44,33 +44,23 @@ public void Build(ISchema schema)
...
-                    var type = context.GetArgument<string>("type");
-
-                    var getCartQuery = new GetCartQuery(storeId, type, cartName, userId, currencyCode, cultureName);
+                    var getCartQuery = context.GetCartQuery<GetCartQuery>();
                     getCartQuery.IncludeFields = context.SubFields.Values.GetAllNodesPaths().ToArray();
                     var cartAggregate = await _mediator.Send(getCartQuery);
                     if (cartAggregate == null)
                     {
-                        var createCartCommand = new CreateCartCommand(storeId, type, cartName, userId, currencyCode, cultureName);
+                        var createCartCommand = new CreateCartCommand(getCartQuery.StoreId, getCartQuery.CartType, getCartQuery.CartName, getCartQuery.UserId, getCartQuery.CurrencyCode, getCartQuery.CurrencyCode);
```
Comment (Vectorfield4) [inline src/XPurchase/VirtoCommerce.XPurchase/Schemas/PurchaseSchema.cs:60]: The last parametr is `getCartQuery.CultureName`.

**Review by Vectorfield4 — CHANGES_REQUESTED** [review]

File: src/XPurchase/VirtoCommerce.XPurchase/Schemas/PurchaseSchema.cs (hunk 1 above)
Comment (Vectorfield4) [reply src/XPurchase/VirtoCommerce.XPurchase/Schemas/PurchaseSchema.cs:60]: image
image

File: src/XPurchase/VirtoCommerce.XPurchase/Authorization/CanAccessCartAuthorizationRequirement .cs [hunk 2]
```diff
@@ -20,25 +21,31 @@ public class CanAccessCartAuthorizationHandler : PermissionAuthorizationHandlerB
...
+                }
 
+                if (result)
+                {
+                    context.Succeed(requirement);
+                }
+                else
+                {
+                    context.Fail();
+                }
```
Comment (yecli) [inline src/XPurchase/VirtoCommerce.XPurchase/Authorization/CanAccessCartAuthorizationRequirement .cs:46]: Doesn't it need to be done outside the if clause - for the admin we need to succeed a requirement?

**Review by yecli — CHANGES_REQUESTED** [review]

File: src/XPurchase/VirtoCommerce.XPurchase/Authorization/CanAccessCartAuthorizationRequirement .cs (hunk 2 above)
Comment (AliveMen) [reply src/XPurchase/VirtoCommerce.XPurchase/Authorization/CanAccessCartAuthorizationRequirement .cs:46]: context.HasSucceeded is true by default. But, yes. Better to move if clause outside

**Review by yecli — APPROVED** [review]

**Review by Vectorfield4 — APPROVED** [review]

**Review by yecli — APPROVED** [review]

**Review by tatarincev — APPROVED** [review]

**Pull request merged**